File: ipymolsysview/widget.py
#!/usr/bin/env python
# coding: utf-8

# Copyright (c) Diego Prada.
# Distributed under the terms of the Modified BSD License.

"""
TODO: Add module docstring
"""

from ipywidgets import DOMWidget
from traitlets import Unicode
from traittypes import Empty, Array
from ._frontend import module_name, module_version

from molsysmt import __version__ as __molsysmt_version__

class Widget(DOMWidget):

    _model_name = Unicode('MolSysModel').tag(sync=True)
    _model_module = Unicode(module_name).tag(sync=True)
    _model_module_version = Unicode(module_version).tag(sync=True)
    _view_name = Unicode('MolSysView').tag(sync=True)
    _view_module = Unicode(module_name).tag(sync=True)
    _view_module_version = Unicode(module_version).tag(sync=True)

    _molsysmt_version = Unicode(__molsysmt_version__).tag(sync=True)

    coordinates = Array(default_value=Empty, dtype=float).tag(sync=True)

    def __init__(self, molecular_system, selection='all', structure_indices='all', syntax='MolSysMT'):

        super(Widget, self).__init__()

        from molsysmt.basic import get

        self.molecular_system = molecular_system
        self.selection = selection
        self.syntax = syntax
        self.structure_indices = structure_indices

        self.n_atoms = get(self.molecular_system, element='atom', selection=selection,
                syntax=syntax, n_atoms=True)

        coordinates = get(self.molecular_system, element='atom', selection=selection,
                syntax=syntax, coordinates=True)

        self.coordinates = Array(puw.get_value(coordinates))

File: js/src/widget.ts
// Copyright (c) Diego Prada
// Distributed under the terms of the Modified BSD License.

import {
  DOMWidgetModel,
  DOMWidgetView,
  ISerializers,
} from '@jupyter-widgets/base';

import { MODULE_NAME, MODULE_VERSION } from './version';

// Import the CSS
import '../css/widget.css';

export class MolSysModel extends DOMWidgetModel {
  defaults() {
    return {
      ...super.defaults(),
      _model_name: MolSysModel.model_name,
      _model_module: MolSysModel.model_module,
      _model_module_version: MolSysModel.model_module_version,
      _view_name: MolSysModel.view_name,
      _view_module: MolSysModel.view_module,
      _view_module_version: MolSysModel.view_module_version,
      value: 'Hello World',
    };
  }

  static serializers: ISerializers = {
    ...DOMWidgetModel.serializers,
    // Add any extra serializers here
  };

  static model_name = 'MolSysModel';
  static model_module = MODULE_NAME;
  static model_module_version = MODULE_VERSION;
  static view_name = 'MolSysView'; // Set to null if no view
  static view_module = MODULE_NAME; // Set to null if no view
  static view_module_version = MODULE_VERSION;
}

export class MolSysView extends DOMWidgetView {
  render() {
    this.el.classList.add('custom-widget');

    this.value_changed();
    this.model.on('change:value', this.value_changed, this);
  }

  value_changed() {
    this.el.textContent = this.model.get('value');
  }
}
